Application/Client/src/app/mapData/mapData.ts

```diff
@@ -34,9 +34,9 @@ export class mapData{
         "Status": "",
         "SubmittedBy": "",
         "SubmittedDate": cardCreatedDate,
-        "IdealFor": "",
-        "Furnished": "",
-        "FuelType": "",
+        "IdealFor": (form.get('IdealFor')!=null)?form.get('IdealFor').value:'',
```

Comment:
Use strict comparison operator !==

Is this is component, do we need to construct json on client side